### ISSUE TITLE
jenkins: mac test failed

### ISSUE THREAD

**Issue by pinggao187** [intel-analytics/analytics-zoo#3729]
issue: [#3719](https://github.com/intel-analytics/analytics-zoo/issues/3719)

I tried to run all packages on mac, and many test cases failed.
jenkins link:
http://10.239.47.210:18888/view/ZOO-NB/job/ZOO-NB-UnitTests-2.4-SCALA-MAC/361/

All Failed Tests
com.intel.analytics.zoo.pipeline.api.NetSpec.Save to tensorflow should works | 2.7 sec | 1
com.intel.analytics.zoo.pipeline.api.NetSpec.graph definition should work | 2.3 sec | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert should be able to work | 0.22 sec | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert with output all blocks should be able to work | 8 ms | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert with configed embedding should be able to work | 8 ms | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.BERT should be able to generate correct result | 7 ms | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert should save/load be able to work | 66 ms | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert gelu should be able to generate correct result | 4 ms | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.Deconvolution2DSpec.Deconvolution2D without bias should be the same as Keras | 4.5 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.AveragePooling1DSpec.AveragePooling1D same mode should be the same as Keras | 4.1 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.Conv1DSpec.Conv1D without bias should be the same as Keras | 4 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.Conv2DSpec.Conv2D NCHW should be the same as Keras | 4.5 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.Conv2DSpec.Conv2D NHWC should be the same as Keras | 4.3 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.Conv2DSpec.Conv2D without bias should be the same as Keras | 4.5 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.DenseSpec.Dense nD input should be the same as Keras | 4 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.GlobalAveragePooling2DSpec.GlobalAveragePooling2D NCHW should be the same as Keras | 4.5 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.GlobalAveragePooling2DSpec.GlobalAveragePooling2D NHWC should be the same as Keras | 4.4 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.GlobalAveragePooling3DSpec.GlobalAveragePooling3D should be the same as Keras | 4.5 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.GlobalMaxPooling2DSpec.GlobalMaxPooling2D NCHW should be the same as Keras | 4 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.GlobalMaxPooling2DSpec.GlobalMaxPooling2D NHWC should be the same as Keras | 4.1 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.GlobalMaxPooling3DSpec.GlobalMaxPooling3D should be the same as Keras | 4 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.MaxPooling1DSpec.MaxPooling1D valid mode should be the same as Keras | 4 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.MaxPooling1DSpec.MaxPooling1D same mode should be the same as Keras | 4.1 sec | 1
com.intel.analytics.zoo.pipeline.api.keras2.layers.Softmax.softmax with 4d input should be the same as Keras | 4.2 sec | 1

**Comment by hkvision** [comment]
```
java.lang.RuntimeException: Nonzero exit value: 1
      at scala.sys.package$.error(package.scala:27)
      at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.slurp(ProcessBuilderImpl.scala:132)
      at scala.sys.process.ProcessBuilderImpl$AbstractBuilder.$bang$bang(ProcessBuilderImpl.scala:102)
      at com.intel.analytics.zoo.pipeline.api.keras.layers.KerasRunner$.run(KerasRunner.scala:153)
      at com.intel.analytics.zoo.pipeline.api.keras.layers.KerasBaseSpec.checkOutputAndGrad(KerasBaseSpec.scala:51)
      at com.intel.analytics.zoo.pipeline.api.keras2.layers.GlobalAveragePooling2DSpec$$anonfun$2.apply$mcV$sp(GlobalAveragePooling2DSpec.scala:58)
      at com.intel.analytics.zoo.pipeline.api.keras2.layers.GlobalAveragePooling2DSpec$$anonfun$2.apply(GlobalAveragePooling2DSpec.scala:45)
      at com.intel.analytics.zoo.pipeline.api.keras2.layers.GlobalAveragePooling2DSpec$$anonfun$2.apply(GlobalAveragePooling2DSpec.scala:45)
```
Most of keras related unit tests seem to have no clear error message...
Could you check keras version? Keras (1.2.2) and Keras2 (2.1.6) should run in different environment

**Comment by pinggao187** [comment]
keras (1.2.2)
http://10.239.47.210:18888/view/ZOO-NB/job/ZOO-NB-UnitTests-2.4-SCALA-MAC/361/

keras(2.1.6)
http://10.239.47.210:18888/job/ZOO-NB-UnitTests-2.4-SCALA-MAC/366/

**Comment by hkvision** [comment]
For keras 1.2.2, all keras 2 unit tests should be disabled, so the failures are:
```
com.intel.analytics.zoo.pipeline.api.NetSpec.Save to tensorflow should works
com.intel.analytics.zoo.pipeline.api.NetSpec.graph definition should work
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert should be able to work
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert with output all blocks should be able to work
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert with configed embedding should be able to work
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.BERT should be able to generate correct result
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert should save/load be able to work
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert gelu should be able to generate correct result
com.intel.analytics.zoo.pipeline.api.keras.layers.Deconvolution2DSpec.Deconvolution2D without bias should be the same as Keras
```
Mostly are BERT related.

**Comment by glorysdj** [comment]
please disable keras 2 tests and enable keras tests

**Comment by pinggao187** [comment]
update

job configure:

mvn clean test -U \
  -DtagsToExclude="com.intel.analytics.zoo.pipeline.api.keras2"
  -Dparallel=false \
  -P spark_2.4+ -Dspark.version=2.4.3 -Dspark-version.project=2.4 \
  -Dbigdl.version=${BIGDL_VERSION} -Ddata-store-url=http://10.239.45.10:8081/repository/raw

All Failed Tests
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert should be able to work | 0.22 sec | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert with output all blocks should be able to work | 8 ms | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert with configed embedding should be able to work | 8 ms | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.BERT should be able to generate correct result | 8 ms | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert should save/load be able to work | 65 ms | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.BertSpec.Bert gelu should be able to generate correct result | 3 ms | 1
com.intel.analytics.zoo.pipeline.api.keras.layers.Deconvolution2DSpec.Deconvolution2D without bias should be the same as Keras

jenkins link:
http://10.239.47.210:18888/view/ZOO-NB/job/ZOO-NB-UnitTests-2.4-SCALA-MAC/368/

**Comment by helenlly** [comment]
@hkvision  @dding3 pls have a look, after enable keras, some cases still failed.

**Comment by dding3** [comment]
For BertSpec failed test, as discussed with @qiuxin2012, after Bigdl-core released, we will update zoo code to use mkl library from bigdl-core instead of zoo-core to fix the failure.

Can we temporarily disable bert ut before bigdl-core release? Or any suggestions?

**Comment by hkvision** [comment]
Does Deconvolution2DSpec pass on linux?

**Comment by pinggao187** [comment]
@hkvision Deconvolution2DSpec pass on linux.

Test name | Duration | Status
-- | -- | --
Deconvolution2D should be the same as Keras | 4.7 sec | Passed
Deconvolution2D without bias should be the same as Keras | 4.9 sec | Passed

jenkins link:
http://10.239.47.210:18888/job/ZOO-NB-UnitTests-2.4-SCALA/949/testReport/

**Comment by hkvision** [comment]
@jason-dai Can we disable Deconvolution2D on mac? This layer is hardly used by users.

**Comment by hkvision** [comment]
Duplicate with #3826 ?

**Comment by dding3** [comment]
BertSpec related test on mac passed on jekins, http://172.168.2.101:8080/view/ZOO-NB/job/ZOO-NB-UnitTests-2.4-SCALA-MAC/381/testReport/com.intel.analytics.zoo.pipeline.api.keras.layers/BertSpec/
